crates/hir_ty/src/lib.rs

```diff
@@ -52,6 +52,12 @@ pub use traits::{InEnvironment, Obligation, ProjectionPredicate, TraitEnvironmen
 
 pub use chalk_ir::{BoundVar, DebruijnIndex};
 
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub enum Lifetime {
+    Parameter(LifetimeParamId),
+    Static,
+}
```

Comment:
@flodiebold any feelings about special-casing the Static lifetime?

Reply:
Yeah, I think it's correct (Chalk does the same thing).